source/Generic/PlayState/PlayState.cs

```diff
@@ -771,29 +773,32 @@ public override void OnGameStopped(OnGameStoppedEventArgs args)
                 }
             }
 
-            RemoveGameStopwatchTuple(game);
+            RemoveGame(game);
         }
 
-        private void CreateGameStopwatchTuple(Game game)
+        private void AddGame(Game game)
         {
-            if (stopwatchList.Any(x => x.Item1 == game.Id))
+            if (playStateData.Any(x => x.Game.Id == game.Id && x.Stopwatch != null))
             {
-                logger.Debug($"A stopwatch for game {game.Name} with id {game.Id} already exists");
+                logger.Debug($"Data for game {game.Name} with id {game.Id} already exists");
             }
             else
             {
-                stopwatchList.Add(new Tuple<Guid, Stopwatch>(game.Id, new Stopwatch()));
-                logger.Debug($"Stopwatch for game {game.Name} with id {game.Id} was created");
+                playStateData.Add(new PlayStateData(game, DateTime.Now, new Stopwatch()));
```

Comment:
Do you think it would make sense to get the `DateTime.Now` value and create the stopwatch inside the class constructor? I think it would make more sense since this is something that doesn't need outside reference and will simplify the object creation

Reply:
Shouldn't be any issue doing so, I will test it.